packages/dapp/src/pages/EscrowInfo.jsx

```diff
@@ -81,19 +83,25 @@ export const EscrowInfo = () => {
   };
 
   const getSmartInvoiceData = async () => {
-    if (
-      utils.isAddress(context.invoice_id) &&
-      !Number.isNaN(parseInt(context.chainID))
-    ) {
-      let smartInvoice = await getSmartInvoiceAddress(
-        context.invoice_id,
-        context.provider
-      );
-      console.log(context.chainID, smartInvoice);
-      getInvoice(parseInt(context.chainID), smartInvoice).then((i) => {
-        console.log(i);
-        setInvoice(i);
-      });
+    try {
+      if (
+        utils.isAddress(context.invoice_id) &&
+        !Number.isNaN(parseInt(context.chainID))
+      ) {
+        let smartInvoice = await getSmartInvoiceAddress(
+          context.invoice_id,
+          context.provider,
+        );
+        console.log(context.chainID, smartInvoice);
+        getInvoice(parseInt(context.chainID), smartInvoice).then(i => {
+          console.log(i);
+          setInvoice(i);
+        });
+      }
+      setInvoiceFetchError(false);
```

Comment:
This maybe called before `getInvoice` finishes. Is that intended?

Reply:
Yes, that's right. Moved that inside the `then` block in the new commit.